### PR TITLE
Multiple submissions per box

### PR DESCRIPTION
**Pull request by dethos**
Closes https://github.com/whitesmith/hawkpost/issues/6

### CODE REVIEW

**Comment by jlbribeiro** [comment]
@dethos:
I found two different issues, one of them which may be addressed later. **Edit:** I added a minor footnote.

---

**Bug**: submitting to a deleted folder throws a 500.
Steps to reproduce:
1. Create a box. No special requirements.
2. Open the box on a new tab.
3. Delete the box on the "main" tab.
4. Submit a message on the "box" tab. You'll get a `DoesNotExist`error:

> Box matching query does not exist.

---

**Bug**: expired boxes appear on the **Open** section if the user never accesses the link of the expired box.
Steps to reproduce:
1. Create a box with a short expiration date.
2. Keep refreshing until you are sure it has expired. You should be able to see the box even though it has expired.
3. Access the link on a different tab. You'll get the expired message.
4. Refresh the main tab again; it should now be gone.

---

On a **footnote:** isn't a blank `expires_at` semantically equivalent to `never_expires`? I can see you changed the `expires_at` type to support empty timestamps ([boxes/models.py#L34](https://github.com/whitesmith/hawkpost/pull/63/files#diff-e7a64c887d63b9895e9644cbb84eb806R34)), that's why I'm asking.

I believe the UI _should remain explicit_ (i.e.: the user should be prompted to click on the "never expires" checkbox,  as a way to acknowledge what it means), but I think it shouldn't need an additional field on the model. From what I could tell (and please correct me if I'm wrong), if I fill the `expires_at` field (by picking a date) and click the "Never expire" checkbox, the `expires_at` field will still be saved, which could be a source for serious bugs in the future (if a developer chooses to write a condition by checking `expires_at` without checking `never_expires` first, for instance).

**tl;dr**: IMHO having to check both fields to make sure the box is still "valid" may be error prone when they are actually _the same thing_.

Also: toggling the "Never expire" checkbox to true/false should disable/enable the date picker.

**Comment by dethos** [comment]
@jlbribeiro I was aware of the second bug, however it is low priority at this moment. It think the best approach would be to create separate issues for each one of the bugs, so they can be addressed individually on other pull requests.

Regarding the footnote, I agree with you and I'll refactor the current PR to make it simpler and more clear in that aspect.

**Comment by jlbribeiro** [comment]
LGTM, merging :+1:
